scripts/zones/AlTaieu/npcs/qm1.lua

```diff
@@ -4,32 +4,28 @@
 -- Allows players to spawn the Jailer of Hope by trading the First Virtue, Deed of Placidity and HQ Phuabo Organ to a ???.
 -- !pos -693 -1 -62 33
 -----------------------------------
-local ID = require("scripts/zones/AlTaieu/IDs");
+local ID = require("scripts/zones/AlTaieu/IDs")
+require("scripts/globals/npc_util")
 -----------------------------------
 
 function onTrade(player,npc,trade)
-    -- JAILER OF HOPE
-    if (
-        not GetMobByID(ID.mob.JAILER_OF_HOPE):isSpawned() and
-        trade:hasItemQty(1850,1) and -- first_virtue
-        trade:hasItemQty(1851,1) and -- deed_of_placidity
-        trade:hasItemQty(1852,1) and -- high-quality_phuabo_organ
-        trade:getItemCount() == 3
-    ) then
-        player:tradeComplete();
-        SpawnMob(ID.mob.JAILER_OF_HOPE):updateClaim(player);
+    if npcUtil.tradeHas(trade, {1850, 1851, 1852}) and npcUtil.popFromQM(player, npc, ID.mob.JAILER_OF_HOPE) then
+        player:confirmTrade()
+        SpawnMob(ID.mob.JAILER_OF_HOPE):updateClaim(player)
```

Comment:
npcUtil.popFromQM already spawns the mob and updates the claim, so this line is redundant.